objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java

```diff
@@ -390,6 +399,362 @@ public void sendDataFromImage(ImageRetrieverResults imageResults, ExtensionServi
        }
    }
    
+   /**
+    * Uploads requested images to VANTIQ. Called when query parameter "operation" is set to "save".
+    * @param request        The parameters sent with the query.
+    * @param replyAddress   The replyAddress used to send a query response.
+    */
+   public void uploadLocalImages(Map<String, ?> request, String replyAddress) {   
```

Comment:
Here, there's a certain amount of duplicated code plus a bunch of mixed processing.  Seems like there might be a better ways to structure things.  Something like:

// Examine parameters
// Check completeness/legality/etc.
// Compute a filter for appropriate parameter set
// Make call to upload_Whatever_ based on filter (and, possibly, if single)

Issue is that right now, there's a very long routine that's doing a whole bunch of different things intermixed.